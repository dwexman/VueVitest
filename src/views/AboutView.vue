<template>
  <div class="about-container">
    <h1>About Page</h1>
    <p>This is the About page!</p>
    <router-link to="/" class="back-link">Back to Home</router-link>
  </div>
</template>

<script>
export default {
  name: 'AboutView',
};
</script>

<style>
.about-container {
  text-align: center;
  padding: 20px;
}

.back-link {
  display: inline-block;
  margin-top: 20px;
  padding: 10px 20px;
  background-color: #28a745;
  color: white;
  text-decoration: none;
  border-radius: 5px;
  transition: background-color 0.3s ease;
}

.back-link:hover {
  background-color: #218838;
}
</style>
