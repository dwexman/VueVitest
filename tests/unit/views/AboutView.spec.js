import { mount } from '@vue/test-utils';
import AboutView from '../../../src/views/AboutView.vue';

describe('AboutView.vue', () => {
  it('renders the About page', () => {
    const wrapper = mount(AboutView);
    expect(wrapper.text()).toContain('About Page');
    expect(wrapper.text()).toContain('This is the About page!');
  });

  it('matches the snapshot', () => {
    const wrapper = mount(AboutView);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
