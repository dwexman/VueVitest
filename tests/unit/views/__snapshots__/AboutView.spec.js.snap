// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutView.vue > matches the snapshot 1`] = `
"<div class="about-container">
  <h1>About Page</h1>
  <p>This is the About page!</p>
  <router-link to="/" class="back-link">Back to Home</router-link>
</div>"
`;
